quartical/gains/tec/__init__.py

```diff
@@ -76,3 +76,121 @@ def make_f_maps(chan_freqs, chan_widths, f_int):
+            np.add.at(
+                ref_data,
+                ant_map,
+                data[sel]
+            )
+            np.add.at(
+                counts,
+                ant_map,
+                flags[sel] == 0 
+            )
+            np.divide(
+                ref_data,
+                counts[:, :, None],
+                where=counts[:, :, None] != 0,
+                out=ref_data
+            )
+
+            for uf in ufint:
+                fsel = np.where(f_map == uf)[0]
+                sel_n_chan = fsel.size
+
+                ###fsel_data = visibilities corresponding to the fsel frequencies
+                fsel_data = ref_data[:, fsel]
+                valid_ant = fsel_data.any(axis=(1, 2))
+                ##in inverse frequency domain
+                invfreq = 1./chan_freq 
+                
+                ##factor for rescaling frequency
+                ffactor = 1e11
+                invfreq *= ffactor
+
+                #delta_freq is the smallest difference between the frequency values
+                delta_freq = invfreq[-2] - invfreq[-1]
+                max_tec = 2 * np.pi/ (delta_freq)
+                super_res = 1000
+                nyq_freq = 1./(2*(invfreq.max()-invfreq.min()))
+                lim0 = 0.5 * max_tec
+
+                ##choosing resolution
+                n = int(super_res * max_tec/ nyq_freq)
+                #domain to pick tec_est from
+                fft_freq = np.linspace(-lim0, lim0, n)
+
+                tec_est = np.zeros((n_ant, n_corr), dtype=np.float64)
+                fft_arr = np.zeros((n_ant, n, n_corr), dtype=fsel_data.dtype)
+
+                for p in range(n_ant): 
+                    for k in range(n_corr):
+                        vis_finufft = finufft.nufft1d3(invfreq, fsel_data[p, :, k], fft_freq, eps=1e-11, isign=-1)
+                        fft_arr[p, :, k] = vis_finufft
+                        fft_data_pk = np.abs(vis_finufft)
+                        tec_est[p, k] = fft_freq[np.argmax(fft_data_pk)]
+
+
+                tec_est[~valid_ant, :] = 0
+
+                for t, p, q in zip(t_map[sel], a1[sel], a2[sel]):
+                    if p == ref_ant:
+                        param[t, uf, q, 0, 1] = -tec_est[q, 0]
+                        if n_corr > 1:
+                            param[t, uf, q, 0, 3] = -tec_est[q, 1]
+                        else:
+                            param[t, uf, p, 0, 1] = tec_est[p, 0]
+                        if n_corr > 1:
+                            param[t, uf, p, 0, 3] = tec_est[p, 1]
+
+                np.save("/home/russeeawon/testing/testing_tecest_quartical/tecest.npy", tec_est)
+                np.save("/home/russeeawon/testing/testing_tecest_quartical/fftarr.npy", fft_arr)
+        
+        import ipdb; ipdb.set_trace()
+        for ut in utint:
+            for f in range(n_chan):
+                fm = f_map[f]
+                cf = 2j * np.pi/ chan_freq[f]
```

Comment:
I may be mistaken but I believe you shouldn't have the factor of 2*pi here because finufft uses angular frequency i.e. omega = 2*pi*freq